extensions/blocks/subscriptions/save.js

```diff
@@ -1,17 +1,128 @@
 	const {
+		subscribePlaceholder,
 		showSubscribersTotal,
-		submitButtonClasses,
-		customBackgroundButtonColor,
-		customTextButtonColor,
+		buttonOnNewLine,
 		submitButtonText,
+		emailFieldBackgroundColor,
+		customEmailFieldBackgroundColor,
+		emailFieldGradient,
+		customEmailFieldGradient,
+		buttonBackgroundColor,
+		customButtonBackgroundColor,
+		buttonGradient,
+		customButtonGradient,
+		textColor,
+		customTextColor,
+		fontSize,
+		customFontSize,
+		borderRadius,
+		borderWeight,
+		borderColor,
+		customBorderColor,
+		padding,
+		spacing,
 	} = attributes;
+
+	const editorSettings = select( 'core/editor' ).getEditorSettings();
+
+	const textColorClass = getColorClassName( 'color', textColor );
+	const fontSizeClass = getFontSizeClass( fontSize );
+	const borderClass = getColorClassName( 'border-color', borderColor );
+
+	const buttonBackgroundClass = getColorClassName( 'background-color', buttonBackgroundColor );
+	const buttonGradientClass = getGradientClass( buttonGradient );
+
+	const emailFieldBackgroundClass = getColorClassName(
+		'background-color',
+		emailFieldBackgroundColor
+	);
+	const emailFieldGradientClass = getGradientClass( emailFieldGradient );
+
+	const sharedClasses = classnames(
+		borderRadius === 0 ? 'no-border-radius' : undefined,
+		fontSizeClass,
+		borderClass
+	);
+
+	const submitButtonClasses = classnames(
+		sharedClasses,
+		textColor ? 'has-text-color' : undefined,
+		textColorClass,
+		buttonBackgroundColor || buttonGradient ? 'has-background' : undefined,
+		buttonBackgroundClass,
+		buttonGradientClass
+	);
+
+	const emailFieldClasses = classnames(
+		sharedClasses,
+		emailFieldBackgroundClass,
+		emailFieldGradientClass
+	);
+
+	const emailFieldBackgroundStyle =
+		! emailFieldBackgroundClass && customEmailFieldGradient
+			? customEmailFieldGradient
+			: customEmailFieldBackgroundColor;
+
+	const buttonBackgroundStyle =
+		! buttonBackgroundClass && customButtonGradient
+			? customButtonGradient
+			: customButtonBackgroundColor;
+
+	const fontSizeStyle = fontSizeClass ? undefined : customFontSize;
+
+	// Themes don't regularly support border color classes, so pass the hex to styles either way.
+	const customBorderColorStyle = getColorObjectByAttributeValues(
+		editorSettings.colors,
+		borderColor,
+		customBorderColor
+	).color;
+
+	const getBlockClassName = () => {
+		return classnames(
+			className,
+			buttonOnNewLine ? 'wp-block-jetpack-subscriptions__newline' : undefined,
+			showSubscribersTotal ? 'wp-block-jetpack-subscriptions__showsubs' : undefined
+		);
+	};
+
 	return (
-		<RawHTML>{ `[jetpack_subscription_form show_only_email_and_button="true" custom_background_button_color="${ customBackgroundButtonColor }" custom_text_button_color="${ customTextButtonColor }" submit_button_text="${ submitButtonText }" submit_button_classes="${ submitButtonClasses }" show_subscribers_total="${ showSubscribersTotal }" ]` }</RawHTML>
+		<div className={ getBlockClassName() }>
```

Comment:
The change in markup will likely invalidate previous uses. If these are necessary we'll need to add a deprecation entry.

Reply:
This is to allow the block to apply its own styles to the widget output while avoiding modifying widget output when not inside the block.

Reply:
@scruffian Does your deprecation work handle this scenario?

Reply:
Confirmed we're good here :+1: